lib/utils/instance.dart

```diff
@@ -9,19 +11,40 @@ String? fetchInstanceNameFromUrl(String? url) {
   return uri.host;
 }
 
-String? generateCommunityInstanceUrl(String? url) {
-  if (url == null) {
-    return null;
+/// Matches instance.tld/c/community@otherinstance.tld
+/// Puts community in group 3 and otherinstance.tld in group 4
+/// https://regex101.com/r/sE8SmL/1
+final RegExp fullCommunityUrl = RegExp(r'^!?(https?:\/\/)?(.*)\/c\/(.*)@(.*)$');
+
+/// Matches instance.tld/c/community
+/// Puts community in group 3 and instance.tld in group 2
+/// https://regex101.com/r/AW2qTr/1
+final RegExp shortCommunityUrl = RegExp(r'^!?(https?:\/\/)?(.*)\/c\/([^@\n]*)$');
+
+/// Matches community@instance.tld
+/// Puts community in group 2 and instance.tld in group 3
+/// https://regex101.com/r/1VrXgX/1
+final RegExp instanceName = RegExp(r'^!?(https?:\/\/)?((?:(?!\/c\/c).)*)@(.*)$');
+
+/// Checks if the given text references a community on a valid Lemmy server.
+/// If so, returns the community name in the format community@instance.tld.
+/// Otherwise, returns null.
+Future<String?> getLemmyCommunity(String text) async {
+  var fullCommunityUrlMatch = fullCommunityUrl.firstMatch(text);
```

Comment:
I'm wondering if we can replace `var` with a specific type here just to reduce the amount of times we use var

I think using var skips out on some of the useful dart's static type checking

Reply:
Done!